osu.Game.Modes.Osu/Objects/Drawables/Pieces/SpinnerDisc.cs

```diff
@@ -0,0 +1,158 @@
+                }
+            };
+        }
+
+        private InputState lastState;
+
+        protected override bool OnMouseDown(InputState state, MouseDownEventArgs args)
+        {
+            lastState = state;
+            return base.OnMouseDown(state, args);
+        }
+
+        protected override bool OnMouseUp(InputState state, MouseUpEventArgs args)
+        {
+            lastState = state;
+            return base.OnMouseUp(state, args);
+        }
+
+        protected override bool OnMouseMove(InputState state)
+        {
+            lastState = state;
+            return base.OnMouseMove(state);
+        }
+        bool tracking;
+        public bool Tracking
+        {
+            get { return tracking; }
+            set
+            {
+                if (value == tracking) return;
+
+                tracking = value;
+            }
+        }
+
+        private bool canCurrentlySpin => Time.Current >= s.StartTime && Time.Current < s.EndTime;
+
+        private float? lastAngle;
+        private float? actualAngle;
+        private float angleAdded;
+        private float lastAngleAdded;
+        private float totalAngleSpinned = 0;
+        public float Progress = 0;
+        private float spinsPerMinuteNeeded = 100 + (5 * 15); //TODO: read per-map OD and place it on the 5
+        private float rotationsNeeded;
+        
+
+        protected override void Update()
+        {
+            
+            base.Update();
+            Progress = MathHelper.Clamp(((totalAngleSpinned / 360) / rotationsNeeded), 0, 1);
+            if (Tracking)
+            {
+                rotationsNeeded = (float)(spinsPerMinuteNeeded * (s.EndTime - s.StartTime) / 60000f);
+                lastAngleAdded = angleAdded;
+                angleAdded += GetAngledDifference();
+                SetAngleSpinned(lastAngleAdded, angleAdded);
+                discScale();
+                disc.RotateTo(Rotation + angleAdded, 500, EasingTypes.OutExpo);
+            }
+            else
+                actualAngle = null;
+            if(!canCurrentlySpin)
+                disc.RotateTo(angleAdded);
+                
+            Tracking = canCurrentlySpin && lastState != null && Contains(lastState.Mouse.NativeState.Position) && lastState.Mouse.HasMainButtonPressed;
+        }
+        private float GetMouseAngledPosition()
+        {
+            float mouseXFromCenter = lastState.Mouse.LastPosition.X - Position.X;
+            float mouseYFromCenter = lastState.Mouse.LastPosition.Y - Position.Y;
+            return (float)MathHelper.RadiansToDegrees(Math.Atan2(mouseYFromCenter,mouseXFromCenter));
+        }
+
+        private void discScale()
+        {
+            disc.ScaleTo(1 + (MathHelper.Clamp(((totalAngleSpinned / 360) / rotationsNeeded), 0, 1)/10), 100);
+        }
+
+        private float GetAngledDifference()
+        {
+            lastAngle = actualAngle ?? GetMouseAngledPosition();
+            Delay(1);

```

Comment:
To get differences in the angle that the mouse is on right now and the angle was last millisecond. I changed this approach locally, but haven't commited yet.